packages/cli-ui/routes/api/folders.js

```diff
@@ -18,4 +18,21 @@ router.get('/', (req, res) => {
   })
 })
 
+// Create new folder
+router.post('/create/', async (req, res) => {
+  const dir = req.query.url
+  if (dir) {
+    if (!fs.existsSync(dir)) {
+      try {
+        await fs.mkdirSync(dir, { recursive: true })
+        res.send('Folder successfully create')
+      } catch (error) {
+        res.send(error)
+      }
+    } else {
+      res.send('Folder already exists')
+    }
+  }
+})
```

Comment:
Я думаю оборачивать надо на верхнем уровне тогда, если возникнет ошибка она отловится во всем коде) И объединить 2 условия можно не зачем плодить if
```suggestion
router.post('/create/', async (req, res) => {
  try {
    const dir = req.query.url
    if (dir && !fs.existsSync(dir)) {
      await fs.mkdirSync(dir, { recursive: true })
      res.send('Folder successfully create')
     } else {
        res.send('Folder already exists')
      }
    }
   } catch (error) {
      res.send(error)
   }
})
```

Reply:
да, нормас